tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py

```diff
@@ -421,23 +421,127 @@ def add_meta_data(net: Element, meta_info: dict):
+        res_data_node = res_node.in_node(0)
+        assert len(res_data_node.in_nodes()) > 0, \
+            "Data node of Result with name {} has no input node.".format(res_node.soft_get('name'))
+        res_in_node = res_data_node.in_node(0)
+
+        # We cannot use out_ports() after port renumbering
+        for v, d in res_in_node.get_sorted_outputs():
+            port_id = d['out'] - len(res_in_node.in_nodes()) if res_in_node.type != 'Const' else d['out']
+            tensor_names = res_in_node.out_port(port_id).get_tensor_names(port_renumber=True)
+            result_names_to_tensor_names[res_node.soft_get('name')] = tensor_names
+    return result_names_to_tensor_names
+
 def serialize_network(graph, net_element, unsupported):
     layers = SubElement(net_element, 'layers')
     edges = SubElement(net_element, 'edges')
     if graph is None:
         return
     nodes = sorted(graph.nodes())
+
+    result_nodes = graph.get_op_nodes(type='Result')
+    result_names_to_tensor_names = get_tensor_names_of_result_node(graph)
+
+    ordered_results = []
+    for output_name in graph.outputs_order:
+        node = graph.get_op_nodes(name=output_name)
+
+        if len(node) == 0:
+            found_tensor_name = False
+            for res_node in result_nodes:
+                res_name = res_node.soft_get('name')
+                tensor_names = result_names_to_tensor_names[res_name]
+                if output_name in tensor_names:
+                    # In this case output tensor name is in tensor names list of previous op
+                    ordered_results.append(res_name)
+                    found_tensor_name = True
+                    break
+
+            if not found_tensor_name:
+                log.warning("Output node with name {} is not found in graph.".format(output_name))
+            continue
+        node = node[0]
+
+        # In this case Result node has the same name as output tensor
+        if node.soft_get('type') == 'Result':
+            ordered_results.append(node.soft_get('name'))
+            continue
+
+        # Here output data node count is checked. Each port cannot have more than one data node.
+        assert len(node.out_nodes()) == 1, "Incorrect graph. Non-Result node with name {} " \
+                                           "has no output data node.".format(output_name)
+
+        # After port renumbering port/connection API is not applicable, and output port numbering
+        # starts from len(node.in_nodes()).
+        data_node = node.out_node(len(node.in_nodes()))
+
+        found_result = False
+        for op_node in data_node.out_nodes():
+            if op_node.soft_get('type') == 'Result':
+                found_result = True
+                ordered_results.append(op_node.soft_get('name'))
+                break
+
+        if not found_result:
+            log.warning("Node that expected to be output with name {} is not connected with Result node.".format(output_name))
```

Comment:
please use new Graph API

Reply:
no way since output ports renumbering (from n, where n is a number of input ports) break graph consistency